### ISSUE TITLE
use new crate-type flag to build cdylib for android

### ISSUE THREAD

**Issue by dvc94ch** [cloudpeers/xbuild#64]
currently it causes cdylibs to be included on all platforms when it's set in Cargo.toml.

**Comment by MarijnS95** [comment]
@dvc94ch Do you think this is something we can insert manually? It can be passed as a flag to `rustc` which perhaps relieves some of the overhead of setting this in `Cargo.toml` manually.

I've also been wanting to figure this out (or find a workaround) for binaries so that you don't need to jump through hoops to make `bin` targets compatible on at least Android, and because a `crate` can have multiple `bin`s but only a single `lib`.

**Comment by dvc94ch** [comment]
Yes. If it's a cdylib on all platforms, set it in Cargo.toml, on android add the crate-type cdylib

**Comment by MarijnS95** [comment]
@dvc94ch I ended up trying this out on `cargo-apk` and it _seems_ to work, shall I port it to `xbuild` as well?

https://github.com/rust-mobile/cargo-apk/pull/28

**Comment by MarijnS95** [comment]
Hmm, given the design of `xbuild` it seems to first call `cargo build` (which we'd have to replace by `cargo rustc --crate-type ...`), followed by platform-specific logic figuring out what would have been built, in order to collect the paths.

It supports a comma-separated list at least, so we'd have to list the build-requirements up-front...
